### DISCUSSION TITLE
Currently type text appears appearing as sole candidate in popup

### DISCUSSION THREAD

**Discussion by doolio** [minad/corfu#439] [Q&A] [answered]
No doubt this is a misconfiguration/misunderstanding on my part. In a `ledger-mode` buffer when I begin to type the name of a new (so I don't expect any completion candidates) multi-word payee the name appears in the corfu popup as a completion candidate. if I hit enter or TAB the candidate is inserted duplicating the first word in the name. This seems to be isolated to `ledger-mode` and from my own testing it seems to only occur for text entered after a date. I presume [`ledger-mode` ](https://github.com/ledger/ledger-mode/blob/cac2b3f34e8f649d7245f7cce8d108a658bb566f/ledger-complete.el#L91) is incorrectly providing the new payee as a candidate.

![image](https://github.com/minad/corfu/assets/13859009/1dd99fb2-4850-4301-a769-28d810220aa1)
![image](https://github.com/minad/corfu/assets/13859009/872e7f6f-9bbd-4992-914c-652d3047340c)

My corfu configuration is as follows:

```emacs-lisp
(use-package corfu                      ; In buffer completion.
  :init (global-corfu-mode)
  :config
  (setq corfu-auto t)
  (setq corfu-quit-no-match t))

(use-feature corfu-history              ; Sorting by history for Corfu.
  :after corfu
  :demand t
  :config (corfu-history-mode))

(use-feature corfu-info  ; Show candidate information in separate buffer.
  :after corfu
  :demand t)

;; If relevant
(use-package ledger-mode                ; Plain text accounting.
  :when (executable-find "ledger")
  :init
  (defun my-ledger-save ()
    "Automatically clean the ledger buffer at each save."
    (interactive)
    (ledger-mode-clean-buffer)
    (whitespace-cleanup)
    (save-buffer)
    (recenter-top-bottom '4))
  :bind (nil
         :map ledger-mode-map
         ("C-x C-s" . my-ledger-save))
  :mode ("\\.dat\\'"
         "\\.ledger\\'"))
```

I appreciate you may not use `ledger-mode` so feel free to ignore.

**Comment by minad** [comment, marked as answer]
Hi! I am sorry, I don't use ledger-mode (yet?). I haven't checked, but it could be that the Ledger Capf can be improved.

**Reply by doolio** [reply]
No worries, and I suspect you're right. If I manage to resolve it I'll report back here. Thanks for your time and all your packages. They're a joy to work with.
